param/__init__.py

```diff
@@ -1628,45 +1628,39 @@ class resolve_path(ParameterizedFunction):
         Prepended to a non-relative path, in order, until a file is
         found.""")
 
-    path_to_file = Boolean(default=True, pickle_default_value=False, doc="""
+    path_to_file = Boolean(default=True, pickle_default_value=False,
+                           allow_None=True, doc="""
         String specifying whether the path refers to a 'File' or a 'Folder'.""")
```

Comment:
I was wondering how it could be anything other than `True` or `False` but looking at the code I see that `None` means *either* a file *or* a directory. I'll clarify this in a small PR.

Minor nitpick but I associate 'folder' with Windows terminology while I consider 'directory' to be the proper platform-independent terminology. Probably not worth updating as everyone knows what you mean!

Reply:
Clarified docstring in #507. I won't mark this as resolved in case you have any opinion on 'Folder' versus 'Directory'.

Reply:
Send like the do string is still confusing as it is a boolean, not a strong as claimed. Thanks for the None clarification. I agree with your preference for "directory" over "folder", but I am just following the existing param.Filename and param.Foldername terminology.